spikeextractors/extractors/neuroscopeextractors/neuroscopeextractors.py

```diff
@@ -0,0 +1,451 @@
+        recording_fn = os.path.join(save_path, RECORDING_NAME)
+
+        # create parameters file if none exists
+        if not os.path.isfile(save_xml):
+            soup = BeautifulSoup("",'xml')
+
+            new_tag = soup.new_tag('nbits')
+            dtype = recording.get_dtype()
+            
+            if not any([dtype == x for x in ['int16', 'int32']]):
+                print('Warning: Unsupported data type passed (',dtype,'); converting to default of int32!',sep="")
+                dtype = 'int32'
+                
+            n_bits = str(dtype)[3:5]
+            new_tag.string = str(n_bits)
+            soup.append(new_tag)
+
+            new_tag = soup.new_tag('nchannels')
+            new_tag.string = str(len(recording.get_channel_ids()))
+            soup.append(new_tag)
+
+            new_tag = soup.new_tag('samplingrate')
+            new_tag.string = str(recording.get_sampling_frequency())
+            soup.append(new_tag)
+
+            # write parameters file
+            f = open(save_xml, "w")
+            f.write(str(soup))
+            f.close()
+            
+        BinDatRecordingExtractor.write_recording(recording, recording_fn, dtype=dtype)
+        
+
+class NeuroscopeSortingExtractor(SortingExtractor):
+
+    """
+    Extracts spiking information from pair of .res and .clu files. The .res is a text file with
+    a sorted list of all spiketimes from all units displayed in sample (integer '%i') units.
+    The .clu file is a file with one more row than the .res with the first row corresponding to
+    the total number of unique ids in the file (and may exclude 0 & 1 from this count)
+    with the rest of the rows indicating which unit id the corresponding entry in the
+    .res file refers to.
+    
+    In the original Neuroscope format:
+        Unit ID 0 is the cluster of unsorted spikes (noise).
+        Unit ID 1 is a cluster of multi-unit spikes.
+        
+    The function defaults to returning multi-unit activity as the first index, and ignoring unsorted noise.
+    To return only the fully sorted units, set keep_mua_units=False.
+        
+    The sorting extractor always returns unit IDs from 1, ..., number of chosen clusters.
+
+    Parameters
+    ----------
+    resfile_path : str
+        Optional. Path to a particular .res text file.
+    clufile_path : str
+        Optional. Path to a particular of .clu text file.
+    folder_path : str
+        Optional. Path to the collection of .res and .clu text files. Will auto-detect format.
+    keep_mua_units : bool
+        Optional. Whether or not to return sorted spikes from multi-unit activity. Defaults to True.
+    """
+    extractor_name = 'NeuroscopeSortingExtractor'
+    installed = True  # check at class level if installed or not
+    is_writable = True
+    mode = 'custom'
+    installation_mesg = ""  # error message when not installed
+
+    def __init__(self, resfile_path=None, clufile_path=None, folder_path=None, keep_mua_units=True):
```

Comment:
Now with the `NeuroscopeMultiSortingExtractor` the `folder_path` arg is not needed here anymore, right? same for all the logic using it

Reply:
Even for the NeuroscopeSortingExtractor, the new folder_path option still allows auto-detection of the .res and .clu files at the location. I've also added the tests for that functionality. It's still back-compatible to the old format as well.